ocaml/libs/xapi-stdext/lib/xapi-stdext-unix/unixext.ml

```diff
@@ -672,6 +672,74 @@ let time_limited_single_read filedesc length ~max_wait =
   in
   Bytes.sub_string buf 0 bytes
 
+(** see [select(2)] "Correspondence between select() and poll() notifications".
+    Note that HUP and ERR are ignored in events and returned only in revents.
+    For simplicity we use the same event mask from the manual in both cases
+ *)
+let pollin_set = Polly.Events.(rdnorm lor rdband lor inp lor hup lor err)
+
+let pollout_set = Polly.Events.(wrband lor wrnorm lor out lor err)
+
+let pollerr_set = Polly.Events.pri
+
+let to_milliseconds ms = ms *. 1e3 |> ceil |> int_of_float
+
+let readable fd (rd, wr, ex) = (fd :: rd, wr, ex)
+
+let writable fd (rd, wr, ex) = (rd, fd :: wr, ex)
```

Comment:
Should this be deduplicated for each list?

Reply:
Is it too expensive to use sets instead of lists (which have problems with duplicates) as the interface?

Reply:
`Unix.select` uses `list`s, and I wanted to keep the interface compatible, so that switching can be done by just using `sed`.

We could change to something else once the select->Unixext.select in https://github.com/xapi-project/xen-api/pull/5911 is done, but that'll require updating the callers too.

Reply:
I've added a comment that we could make this data structure change later.